matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/handler/UserAccountDataSyncHandler.kt

```diff
@@ -81,20 +82,24 @@ internal class UserAccountDataSyncHandler @Inject constructor(
 
     fun handle(realm: Realm, accountData: UserAccountDataSync?) {
         accountData?.list?.forEach { event ->
-            // Generic handling, just save in base
-            handleGenericAccountData(realm, event.type, event.content)
-            when (event.type) {
-                UserAccountDataTypes.TYPE_DIRECT_MESSAGES -> handleDirectChatRooms(realm, event)
-                UserAccountDataTypes.TYPE_PUSH_RULES -> handlePushRules(realm, event)
-                UserAccountDataTypes.TYPE_IGNORED_USER_LIST -> handleIgnoredUsers(realm, event)
-                UserAccountDataTypes.TYPE_BREADCRUMBS -> handleBreadcrumbs(realm, event)
+            if (event.content.isEmpty()) {
+                UserAccountDataEntity.delete(realm, event.type)
```

Comment:
Should we do some treatment on specific Account data as per the `when` block in the `else` block below?

So I would keep the existing code, and move the removal from DB to the `handleGenericAccountData()` method.